imports/server/mediasoup.ts

```diff
@@ -0,0 +1,763 @@
+    transport.observer.on('newproducer', Meteor.bindEnvironment(this.onProducerCreated.bind(this)));
+    transport.observer.on('newconsumer', Meteor.bindEnvironment(this.onConsumerCreated.bind(this)));
+
+    transport.observer.on('close', Meteor.bindEnvironment(() => {
+      this.transports.delete(`${transport.appData.peer}:${transport.appData.direction}`);
+      Transports.remove({ transportId: transport.id });
+      TransportStates.remove({ transportId: transport.id });
+    }));
+
+    if (!(transport instanceof types.WebRtcTransport)) {
+      Ansible.warn('Ignoring unexpected non-WebRTC transport', { call: transport.appData.call, peer: transport.appData.peer, transport: transport.id });
+      return;
+    }
+
+    transport.observer.on('icestatechange', Meteor.bindEnvironment((iceState: types.IceState) => {
+      TransportStates.upsert({
+        createdServer: serverId,
+        transportId: transport.id,
+      }, {
+        $set: {
+          iceState,
+          createdBy: transport.appData.createdBy,
+        },
+      });
+    }));
+    transport.observer.on('iceselectedtuplechange', Meteor.bindEnvironment((iceSelectedTuple?: types.TransportTuple) => {
+      TransportStates.upsert({
+        createdServer: serverId,
+        transportId: transport.id,
+      }, {
+        $set: {
+          iceSelectedTuple: iceSelectedTuple ? JSON.stringify(iceSelectedTuple) : undefined,
+          createdBy: transport.appData.createdBy,
+        },
+      });
+    }));
+    transport.observer.on('dtlsstatechange', Meteor.bindEnvironment((dtlsState: types.DtlsState) => {
+      TransportStates.upsert({
+        createdServer: serverId,
+        transportId: transport.id,
+      }, {
+        $set: {
+          dtlsState,
+          createdBy: transport.appData.createdBy,
+        },
+      });
+    }));
+
+    this.producers.forEach((producer) => {
+      this.createConsumer(transport.appData.direction, transport, producer);
+    });
+
+    this.transports.set(`${transport.appData.peer}:${transport.appData.direction}`, transport);
+    Transports.insert({
+      call: transport.appData.call,
+      createdServer: serverId,
+      peer: transport.appData.peer,
+      transportRequest: transport.appData.transportRequest,
+      direction: transport.appData.direction,
+      transportId: transport.id,
+      iceParameters: JSON.stringify(transport.iceParameters),
+      iceCandidates: JSON.stringify(transport.iceCandidates),
+      dtlsParameters: JSON.stringify(transport.dtlsParameters),
+      createdBy: transport.appData.createdBy,
+    });
+  }
+
+  onProducerCreated(producer: types.Producer) {
+    producer.observer.on('close', Meteor.bindEnvironment(() => {
+      this.producers.delete(producer.appData.producerClient);
+      ProducerServers.remove({ producerId: producer.id });
+    }));
+
+    // Create consumers for other existing transports
+    this.transports.forEach(async (transport, key) => {
```

Comment:
This too needs to get filtered down to just the call

Reply:
(Also fixed in 1609128)